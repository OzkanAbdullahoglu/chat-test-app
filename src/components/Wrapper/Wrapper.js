/* eslint-disable linebreak-style */
import styled from 'styled-components';

export const Wrapper = styled.div`
  width: ${(props) =>
    props.incomingIcon || props.incomingAvatar || props.sendIcon
      ? '40px'
      : '20px'};
  height: ${(props) =>
    props.incomingIcon || props.incomingAvatar || props.sendIcon
      ? '40px'
      : '20px'};
  display: flex;
  opacity: ${(props) => (props.hideButton ? 0 : 1)};
  transition-property: opacity;
  transition-duration: 0.5s;
  transition-timing-function: linear;
  transition-delay: 0.1s;
  position: ${(props) =>
    props.sendIcon ? 'absolute' : props.scrollDown ? 'sticky' : 'relative'};
  border-radius: 50%;
  justify-content: center;
  align-items: center;
  color: #fff;
  background-color: ${(props) => (props.incomingIcon ? ' #fff' : '#00bcd4')};
  border: ${(props) => (props.incomingIcon ? '1px solid #00bcd4' : 'none')};
  cursor: ${(props) =>
    props.scrollDown || props.incomingIcon || props.sendIcon
      ? 'pointer'
      : 'auto'};
  margin: ${(props) => (props.scrollDown ? '0 10px 0 0' : '0 7px 0 10px')};
  right: ${(props) => (props.sendIcon || props.scrollDown ? '0' : 'inherit')};
  bottom: ${(props) => (props.sendIcon ? '10px' : 'inherit')};
  top: ${(props) =>
    props.scrollDown ? '625px' : props.sendIcon ? '10px' : '0'};
  left: ${(props) => (props.scrollDown ? '410px' : 'inherit')};
  box-shadow: 0.1em 0.1em rgba(0, 0, 0, 0.12);
  -moz-box-shadow: 0.1em 0.1em rgba(0, 0, 0, 0.12);
  -webkit-box-shadow: 0.1em 0.1em rgba(0, 0, 0, 0.12);
  -o-box-shadow: 0.1em 0.1em rgba(0, 0, 0, 0.12);
  z-index: 2;
  &:hover {
    background: ${(props) =>
    props.scrollDown || props.incomingIcon ? '#00bcd4' : '#00bcd4'};
    color: #fff;
  }
  &:focus {
    outline: '1px solid black';
    outline-offset: -4px;
  }

  @media (max-height: 1370px) and (max-width: 1024px) {
    top: ${(props) =>
    props.scrollDown ? '1250px' : props.sendIcon ? '10px' : '0'};
    left: ${(props) => (props.scrollDown ? '390px' : 'inherit')};
  }

  @media (max-height: 1024px) and (max-width: 770px) {
    top: ${(props) =>
    props.scrollDown ? '900px' : props.sendIcon ? '10px' : '0'};
    left: ${(props) => (props.scrollDown ? '380px' : 'inherit')};
  }
  @media (max-height: 823px) and (max-width: 415px) {
    top: ${(props) =>
    props.scrollDown ? '600px' : props.sendIcon ? '10px' : '0'};
    left: ${(props) => (props.scrollDown ? '310px' : 'inherit')};
  }
  @media (max-height: 670px) and (max-width: 375px) {
    top: ${(props) =>
    props.scrollDown ? '500px' : props.sendIcon ? '10px' : '0'};
    left: ${(props) => (props.scrollDown ? '300px' : 'inherit')};
  }
  @media (max-height: 640px) and (max-width: 360px) {
    top: ${(props) =>
    props.scrollDown ? '475px' : props.sendIcon ? '10px' : '0'};
    left: ${(props) => (props.scrollDown ? '300px' : 'inherit')};
  }

  @media (max-height: 570px) and (max-width: 320px) {
    top: ${(props) =>
    props.scrollDown ? '440px' : props.sendIcon ? '10px' : '0'};
    left: ${(props) => (props.scrollDown ? '250px' : 'inherit')};
  }
`;
