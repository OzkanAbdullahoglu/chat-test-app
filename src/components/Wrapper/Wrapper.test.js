/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import { Wrapper } from './Wrapper';

describe('Wrapper', () => {
  const props = {
    incomingIcon: true,
    incomingAvatar: true,
    sendIcon: true,
    hideButton: true,
    scrollDown: true,
  };

  it('renders without crashing', () => {
    const wrapper = shallow(<Wrapper {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });


  it('renders properly under default props', () => {
    const wrapper = shallow(<Wrapper {...props} />);
    expect(wrapper).toHaveStyleRule('height', '40px');
    expect(wrapper).toHaveStyleRule('width', '40px');
    expect(wrapper).toHaveStyleRule('background-color', '#fff');
    expect(wrapper).toHaveStyleRule('margin', '0 10px 0 0');
    expect(wrapper).toHaveStyleRule('position', 'absolute');
    expect(wrapper).toHaveStyleRule('cursor', 'pointer');
    expect(wrapper).toHaveStyleRule('opacity', '0');
    expect(wrapper).toHaveStyleRule('border', '1px solid #00bcd4');
    expect(wrapper).toHaveStyleRule('right', '0');
    expect(wrapper).toHaveStyleRule('bottom', '10px');
    expect(wrapper).toHaveStyleRule('top', '625px');
    expect(wrapper).toHaveStyleRule('left', '410px');
  });
  it('renders properly under custom props', () => {
    const customProps = {
      incomingIcon: false,
      incomingAvatar: false,
      sendIcon: false,
      hideButton: false,
      scrollDown: false,
    };
    const wrapper = shallow(<Wrapper {...customProps} />);
    expect(wrapper).toHaveStyleRule('height', '20px');
    expect(wrapper).toHaveStyleRule('width', '20px');
    expect(wrapper).toHaveStyleRule('background-color', '#00bcd4');
    expect(wrapper).toHaveStyleRule('margin', '0 7px 0 10px');
    expect(wrapper).toHaveStyleRule('position', 'relative');
    expect(wrapper).toHaveStyleRule('cursor', 'auto');
    expect(wrapper).toHaveStyleRule('opacity', '1');
    expect(wrapper).toHaveStyleRule('border', 'none');
    expect(wrapper).toHaveStyleRule('right', 'inherit');
    expect(wrapper).toHaveStyleRule('bottom', 'inherit');
    expect(wrapper).toHaveStyleRule('top', '0');
    expect(wrapper).toHaveStyleRule('left', 'inherit');
  });
});
