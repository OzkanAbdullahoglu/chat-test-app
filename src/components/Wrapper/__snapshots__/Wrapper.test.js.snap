// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper renders without crashing 1`] = `
.c0 {
  width: 40px;
  height: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 0;
  -webkit-transition-property: opacity;
  transition-property: opacity;
  -webkit-transition-duration: 0.5s;
  transition-duration: 0.5s;
  -webkit-transition-timing-function: linear;
  transition-timing-function: linear;
  -webkit-transition-delay: 0.1s;
  transition-delay: 0.1s;
  position: absolute;
  border-radius: 50%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  background-color: #fff;
  border: 1px solid #00bcd4;
  cursor: pointer;
  margin: 0 10px 0 0;
  right: 0;
  bottom: 10px;
  top: 625px;
  left: 410px;
  box-shadow: 0.1em 0.1em rgba(0,0,0,0.12);
  -moz-box-shadow: 0.1em 0.1em rgba(0,0,0,0.12);
  -webkit-box-shadow: 0.1em 0.1em rgba(0,0,0,0.12);
  -o-box-shadow: 0.1em 0.1em rgba(0,0,0,0.12);
  z-index: 2;
}

.c0:hover {
  background: #00bcd4;
  color: #fff;
}

.c0:focus {
  outline: '1px solid black';
  outline-offset: -4px;
}

@media (max-height:1370px) and (max-width:1024px) {
  .c0 {
    top: 1250px;
    left: 390px;
  }
}

@media (max-height:1024px) and (max-width:770px) {
  .c0 {
    top: 900px;
    left: 380px;
  }
}

@media (max-height:823px) and (max-width:415px) {
  .c0 {
    top: 600px;
    left: 310px;
  }
}

@media (max-height:670px) and (max-width:375px) {
  .c0 {
    top: 550px;
    left: 300px;
  }
}

@media (max-height:640px) and (max-width:360px) {
  .c0 {
    top: 500px;
    left: 300px;
  }
}

@media (max-height:570px) and (max-width:320px) {
  .c0 {
    top: 440px;
    left: 250px;
  }
}

<div
  className="c0"
/>
`;
